src/frequenz/sdk/util/ringbuffer.py

```diff
@@ -0,0 +1,111 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with focus on time & memory efficiency"""
+
+from typing import TypeVar, Generic, Sequence
+from abc import ABC, abstractmethod
+
+from collections import deque
+
+from enum import Enum
+
+import numpy as np
+from itertools import chain
+
+
+T = TypeVar('T')
+
+class AbstractRingBufferContainer(Generic[T], ABC):
+    def __init__(self, size):
+        self.size = size
+
```

Comment:
The size here is basically the max size, as it is a non-growing container that we pass to the ring queue

Reply:
I don't think we need a max-size,  because these will be fixed-size buffers and users are required to specify a size every time they create an instance,  so we're eliminating the possibility of it using up all memory without the developer being aware of it or having no control over it.